Release notes - 5.2.0-preview2 (#2056)
dotnet/SqlClient: release-notes/5.2/5.2.0-preview2.md

# Release Notes

## [Preview Release 5.2.0-preview2.23159.1] - 2023-06-08

This update brings the below changes over the previous release:

### Added

- Added new property `RowsCopied64` to `SqlBulkCopy`. [#2004](https://github.com/dotnet/SqlClient/pull/2004) [Read more](#added-new-property-rowscopied64-to-sqlbulkcopy)

### Fixed

> - Fixed mixed synchronization issue in managed SNI. [#1029](https://github.com/dotnet/SqlClient/pull/1029)

Reviewer: ```suggestion
- Fixed socket synchronization issue during connect in managed SNI. [#1029](https://github.com/dotnet/SqlClient/pull/1029)
```

- Fixed issue with `SqlConnectionStringBuilder` property indexer not supporting non-string values. [#2018](https://github.com/dotnet/SqlClient/pull/2018)

### Changed

- Moved to new System.Data.SqlTypes APIs in **.NET 7** and upper. [1934](https://github.com/dotnet/SqlClient/pull/1934) and [#1981](https://github.com/dotnet/SqlClient/pull/1981)
- Changed **[UseOneSecFloorInTimeoutCalculationDuringLogin](https://learn.microsoft.com/sql/connect/ado-net/appcontext-switches#enable-a-minimum-timeout-during-login)** App Context switch default to **true** and extended its effect to .NET and .NET Standard. [#2012](https://github.com/dotnet/SqlClient/pull/2012)
- Updated `Microsoft.Identity.Client` version from 4.47.2 to 4.53.0. [#2031](https://github.com/dotnet/SqlClient/pull/2031), [#2055](https://github.com/dotnet/SqlClient/pull/2055) 
- Code health improvement: [#1985](https://github.com/dotnet/SqlClient/pull/1985)

## New features over preview release v5.2.0-preview1

### Added new property `RowsCopied64` to SqlBulkCopy

SqlBulkCopy has a new property `RowsCopied64` which supports `long` value types.

> **Note that the existing SqlBulkCopy.RowsCopied behaviour of going negative is still preserved for `int` types.**

Reviewer: ```suggestion
**Note that the existing `SqlBulkCopy.RowsCopied` behavior is unchanged. When the value exceeds `int.MaxValue`, `RowsCopied` can return a negative number.**
```

# Release Notes

## [Preview Release 5.2.0-preview2.x] - 2023-06-08

This update brings the below changes over the previous release:

### Added

> - Added new property `RowsCopied64` to `SqlBulkCopy`. [#2004](https://github.com/dotnet/SqlClient/pull/2004)

Reviewer 2: Could you add a sample for the new API?

### Fixed

- Fixed mixed synchronization issue in managed SNI. [#1029](https://github.com/dotnet/SqlClient/pull/1029)
- Fixed issue with `SqlConnectionStringBuilder` property indexer not supporting non-string values. [#2018](https://github.com/dotnet/SqlClient/pull/2018)

### Changed

> - Moved to new System.Data.SqlTypes APIs in **.NET 7** and **.NET 8**. [1934](https://github.com/dotnet/SqlClient/pull/1934) and [#1981](https://github.com/dotnet/SqlClient/pull/1981)

Reviewer 2: ```suggestion
- Moved to new System.Data.SqlTypes APIs in **.NET 7** and upper. [1934](https://github.com/dotnet/SqlClient/pull/1934) and [#1981](https://github.com/dotnet/SqlClient/pull/1981)
```

# Release Notes

## [Preview Release 5.2.0-preview2.x] - 2023-06-08

This update brings the below changes over the previous release:

### Added

- Added SQL Types compatibility with **.NET 8**. [1934](https://github.com/dotnet/SqlClient/pull/1934)
- Added support for **.NET 7.0**. [#1981](https://github.com/dotnet/SqlClient/pull/1981)

> - Added new property 'RowsCopied64' to SqlBulkCopy. [#2004](https://github.com/dotnet/SqlClient/pull/2004)

Reviewer 3: Ticks seem wrong here

Author: is your concern addressed by removing the tick from SqlBulkCopy?

Reviewer: He's saying they are the wrong ticks/quote character. Should be backquote instead of forward quote. They have been changed since the original commit, so this is resolved.

> ### Added
> 
> - Added SQL Types compatibility with **.NET 8**. [1934](https://github.com/dotnet/SqlClient/pull/1934)
> - Added support for **.NET 7.0**. [#1981](https://github.com/dotnet/SqlClient/pull/1981)
> - Added new property 'RowsCopied64' to SqlBulkCopy. [#2004](https://github.com/dotnet/SqlClient/pull/2004)
> - Added minimum login timeout as 1 second as the default behavior on **.NET Core**. [#2012](https://github.com/dotnet/SqlClient/pull/2012)
> 
> ### Fixed
> 
> - Fixed the mixed synchronization issue in managed SNI. [#1029](https://github.com/dotnet/SqlClient/pull/1029)
> - Fixed the issue with SqlConnectionStringBuilder property indexer not supporting non-string values. [#2018](https://github.com/dotnet/SqlClient/pull/2018)
> 
> ### Changed
> 
> - Updated Microsoft.Identity.Client version from 4.47.2 to 4.53.0. [#2031](https://github.com/dotnet/SqlClient/pull/2031), [#2055](https://github.com/dotnet/SqlClient/pull/2055) 
> - Code health improvements: [#1985](https://github.com/dotnet/SqlClient/pull/1985)

Reviewer: ```suggestion
### Added

- Added new property `RowsCopied64` to `SqlBulkCopy`. [#2004](https://github.com/dotnet/SqlClient/pull/2004)

### Fixed

- Fixed mixed synchronization issue in managed SNI. [#1029](https://github.com/dotnet/SqlClient/pull/1029)
- Fixed issue with `SqlConnectionStringBuilder` property indexer not supporting non-string values. [#2018](https://github.com/dotnet/SqlClient/pull/2018)

### Changed

- Moved to new System.Data.SqlTypes APIs in **.NET 7** and **.NET 8**. [1934](https://github.com/dotnet/SqlClient/pull/1934) and [#1981](https://github.com/dotnet/SqlClient/pull/1981)
- Changed **[UseOneSecFloorInTimeoutCalculationDuringLogin](https://learn.microsoft.com/sql/connect/ado-net/appcontext-switches#enable-a-minimum-timeout-during-login)** App Context switch default to **true** and extended its effect to .NET and .NET Standard. [#2012](https://github.com/dotnet/SqlClient/pull/2012)
- Updated `Microsoft.Identity.Client` version from 4.47.2 to 4.53.0. [#2031](https://github.com/dotnet/SqlClient/pull/2031), [#2055](https://github.com/dotnet/SqlClient/pull/2055) 
- Code health improvements: [#1985](https://github.com/dotnet/SqlClient/pull/1985)
```

> - Code health improvements: [#1985](https://github.com/dotnet/SqlClient/pull/1985)

Reviewer 2: ```suggestion
- Code health improvement: [#1985](https://github.com/dotnet/SqlClient/pull/1985)
```
